### PR TITLE
Chat Diorama

### PR DESCRIPTION
**Pull request by ultrabot2022**
Hi @lalalune 
I fixed the issue where the player's chat diorama would go out of frame when crouching or riding in a vehicle.
Please check it and give me feedback.

### CODE REVIEW

File: src/CharacterHups.jsx [hunk 1]
```diff
@@ -1,13 +1,13 @@
 // import * as THREE from 'three';
-import React, {useState, useEffect, useRef} from 'react';
-import classnames from 'classnames';
-import dioramaManager from '../diorama.js';
-import {RpgText} from './RpgText.jsx';
-import styles from './CharacterHups.module.css';
+import React, { useState, useEffect, useRef } from "react";
```
Comment (lalalune) [inline src/CharacterHups.jsx:2]: Please Make sure you keep the curlies and quotes the same as they were, and follow the format for these for additions as well. Really hard to read the PR when there is so much changed that is just formatting.

File: src/CharacterHups.jsx [hunk 2]
```diff
@@ -34,16 +34,27 @@ const CharacterHup = function(props) {
     if (canvasRef.current) {
       const canvas = canvasRef.current;
       const player = hup.parent.player;
-      let {diorama, avatar} = chatDioramas.get(player) ?? {diorama: null, avatar: null};
+      let { diorama, avatar } = chatDioramas.get(player) ?? {
```
Comment (lalalune) [inline src/CharacterHups.jsx:37]: Again, if this hasn't changed, please change all formatting back

File: src/CharacterHups.jsx [hunk 3]
```diff
@@ -34,16 +34,27 @@ const CharacterHup = function(props) {
...
+        diorama: null,
+        avatar: null,
+      };
       if (diorama && player.avatar.model === avatar) {
         // console.log('got diorama', diorama);
         diorama.resetCanvases();
         diorama.addCanvas(canvas);
       } else {
-        avatar = player.avatar.model;
+        let neckBone;
```
Comment (lalalune) [inline src/CharacterHups.jsx:46]: This seems to be the major change of this PR, can we isolate just to that?

**Comment by lalalune** [comment]
@ultrabot2022 please edit your comment to reference the Github issues this fixes and record a video of it working so QA knows what to look for.

**Comment by alisaad673** [comment]
@ultrabot2022 I tried it but the chat diorama and pfp, both are being viewed from the backside of the character. Thus back of the head is visible and not the face in both pfp and diorama. Can you fix this plz


https://user-images.githubusercontent.com/35926530/181254813-6eb4019f-f056-40a1-a6ae-85f53479218e.mp4

**Comment by ultrabot2022** [comment]
https://user-images.githubusercontent.com/80860637/181345857-e4f37334-b13e-4226-8d9d-8ee394a066f8.mp4

I fixed a chat diorama going out issue when a user sent a chat.
It seems the video that @alisaad673 provided has other issue, and I'll fix it soon.

**Comment by ultrabot2022** [comment]
https://user-images.githubusercontent.com/80860637/181352926-27da133c-e1e6-4317-b166-238226014477.mp4

**Comment by lalalune** [comment]
This looks good but probably should run by @avaer to make sure the framing is what he had in mind

**Comment by lalalune** [comment]
@ultrabot2022 looks good but left some comments, we should try to avoid making any new allocations

**Comment by alisaad673** [comment]
@ultrabot2022  
The pfp is still being shown from the backside. Also you need to rebase it over to master i guess as its on an older version and a few prs were reverted from that version. Also the street scene is empty in your branch.

**Comment by ultrabot2022** [comment]
https://user-images.githubusercontent.com/80860637/181662471-c789cd84-8069-4f34-a82f-f3c13ef4b06e.mp4

@alisaad673. This function is working well on my side. Please check again.

**Comment by ultrabot2022** [comment]
https://user-images.githubusercontent.com/80860637/181662471-c789cd84-8069-4f34-a82f-f3c13ef4b06e.mp4

@alisaad673. This function is working well on my side. Please check again.

**Comment by ultrabot2022** [comment]
https://user-images.githubusercontent.com/80860637/181662471-c789cd84-8069-4f34-a82f-f3c13ef4b06e.mp4

@alisaad673. This function is working well on my side. Please check again.

**Comment by ultrabot2022** [comment]
https://user-images.githubusercontent.com/80860637/181662471-c789cd84-8069-4f34-a82f-f3c13ef4b06e.mp4

@alisaad673. This function is working well on my side. Please check again.

**Review by lalalune — APPROVED** [review]

**Pull request merged**